### PR TITLE
[TypeSpec-Clean]: Updating HealthInsights

### PR DESCRIPTION
**Pull request by markcowl**
- TypeSpecValidation is failing due to a bug in the tool (validates all folders under `cognitiveservices`).
- TypeSpecApiView is failing because it tries to generate swagger from folders like `healthinsights.oncophenotype` that are not configured to generate Swagger.
- Passing CI: https://dev.azure.com/azure-sdk/public/_build/results?buildId=2772654&view=logs&j=9cd39ef8-47c1-5b58-08db-8c3708712f1b&t=467c1474-02d7-5bdd-89c6-4f3a05c439d0

### CODE REVIEW

File: specification/cognitiveservices/HealthInsights/healthinsights.openapi/tspconfig.yaml [hunk 1]
```diff
@@ -6,5 +6,7 @@ emit: [
 
 options:
   "@azure-tools/typespec-autorest":
-     output-file: "openapi.json"
+     azure-resource-provider-folder: "HealthInsights"
+     emitter-output-dir: "{project-root}/../../data-plane"
+     output-file: "{azure-resource-provider-folder}/{version-status}/{version}/openapi.json"
```
Comment (mikeharder) [inline specification/cognitiveservices/HealthInsights/healthinsights.openapi/tspconfig.yaml:11]:
```suggestion
     azure-resource-provider-folder: "./data-plane"
     emitter-output-dir: "{project-root}/../.."
     output-file: "{azure-resource-provider-folder}/{service-name}/{version-status}/{version}/openapi.json"
```

Would this be better?  I think this is equivalent and it aligns with `Contoso.WidgetManager` (assuming the sample is most correct).

File: specification/cognitiveservices/HealthInsights/healthinsights.trialmatcher/route.trialmatcher.tsp [hunk 2]
```diff
@@ -17,15 +17,21 @@ interface TrialMatcher {
...
   @tag("TrialMatcher")
   @doc("Gets the status and details of the Trial Matcher job.")
-  @example("./examples/SuccessfulTrialMatcherResponse.json", "SuccessfulTrialMatcherGetAnalyzeStatus")
+  @example(
+    "./examples/SuccessfulTrialMatcherResponse.json",
+    "SuccessfulTrialMatcherGetAnalyzeStatus"
+  )
   GetJob is Azure.Core.ResourceRead<TrialMatcherResult>;
 
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "There is no long-runnign RPC template in Azure.Core"
```
Comment (mikeharder) [inline specification/cognitiveservices/HealthInsights/healthinsights.trialmatcher/route.trialmatcher.tsp:26]:
```suggestion
  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "There is no long-running RPC template in Azure.Core"
```

File: specification/cognitiveservices/HealthInsights/healthinsights.oncophenotype/route.oncophenotype.tsp [hunk 3]
```diff
@@ -17,14 +17,21 @@ interface OncoPhenotype {
...
   @tag("OncoPhenotype")
   @doc("Gets the status and details of the Onco Phenotype job.")
-  @example("./examples/SuccessfulOncoPhenotypeResponse.json", "SuccessfulOncoPhenotypeGetAnalyzeStatus")
+  @example(
+    "./examples/SuccessfulOncoPhenotypeResponse.json",
+    "SuccessfulOncoPhenotypeGetAnalyzeStatus"
+  )
   GetJob is Azure.Core.ResourceRead<OncoPhenotypeResult>;
 
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "There is no long-runnign RPC template in Azure.Core"
```
Comment (mikeharder) [inline specification/cognitiveservices/HealthInsights/healthinsights.oncophenotype/route.oncophenotype.tsp:26]:
```suggestion
  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "There is no long-running RPC template in Azure.Core"
```

**Review by mikeharder — CHANGES_REQUESTED** [review]

**Review by mikeharder — APPROVED** [review]

**Review by mikeharder — APPROVED** [review]

**Comment by markcowl** [comment]
@mikeharder I made one change - the service name is set from the root namespace, which does not match where this swagger should be emitted.  I used the literal service name rather than the interpolated variable, rather than changing the namespace.

**Comment by mikeharder** [comment]
> @mikeharder I made one change - the service name is set from the root namespace, which does not match where this swagger should be emitted. I used the literal service name rather than the interpolated variable, rather than changing the namespace.

Should we change the namespace instead?

I'm fine if we want to fix this in a later PR, and for now only make changes required to successfully build the folder.  But longer-term I think we want to align all the existing specs folders in the repo as much as we can.

So if the convention is the swagger should be emitted to a folder named the same as the namespace, I'd like to make this consistent across specs.

**Pull request merged**